### PR TITLE
Ported sell function. Add unit tests. Create sell route. Add integration tests

### PR DESCRIPTION
**Pull request by TheHollidayInn**
#6887 User ops sell
### Changes

Ported sell. 
Added unit tests. 
Add sell route. 
Added integration tests

---

UUID: 206039c6-24e4-4b9f-8a31-61cbb9aa3f66

### CODE REVIEW

File: common/script/ops/sell.js [hunk 1]
```diff
@@ -1,23 +1,41 @@
...
+
+  if (!type) {
+    throw new BadRequest(i18n.t('typeRequired', req.language));
+  }
+
+  if (!key) {
+    throw new BadRequest(i18n.t('keyRequired', req.language));
+  }
+
+  let acceptedTypes = ['eggs', 'hatchingPotions', 'food'];
```
Comment (crookedneighbor) [inline common/script/ops/sell.js]: I think this should be a constant declared outside of the function.

Comment (TheHollidayInn) [inline common/script/ops/sell.js]: I agree. Do we have a good place to put constants?

Comment (crookedneighbor) [inline common/script/ops/sell.js]: Just put it at the top of this file under the imports.

Comment (TheHollidayInn) [inline common/script/ops/sell.js]: Oh.. I read that as outside of the file not outside the function. My bad ha

**Comment by paglias** [comment]
The file should not be excluded anymore from linting

**Comment by TheHollidayInn** [comment]
Updated!

**Pull request merged**